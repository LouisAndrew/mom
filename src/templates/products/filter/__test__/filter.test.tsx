import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Filter from '..';

describe('Filter component', () => {
    const el: React.ReactElement = (
        <Filter
            locationOptions={[]}
            saleTypeOptions={[]}
            propTypeOptions={[]}
            addressFilter={''}
            price={{ min: 0, max: 0 }}
            area={{ min: 0, max: 0 }}
            handleChangeAddress={() => {
                return;
            }}
            handleSelectLocations={() => {
                return;
            }}
            handleSelectSaleType={() => {
                return;
            }}
            handleSelectPropertyType={() => {
                return;
            }}
            handleChangeArea={() => {
                return;
            }}
            handleChangePrice={() => {
                return;
            }}
            applyFilters={() => {
                return;
            }}
            collapseFilter={() => {
                return;
            }}
        />
    );

    afterEach(cleanup);
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
