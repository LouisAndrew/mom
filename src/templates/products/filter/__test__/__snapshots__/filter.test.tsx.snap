// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component matches snapshot 1`] = `
<div
  className="sc-fzqBZW cZeasb"
  width={
    Array [
      "90%",
      "90%",
      "fit-content",
    ]
  }
>
  <span
    role="button"
    tabIndex={0}
  >
    <h2
      className="sc-AxirZ kopktq"
      color={
        Array [
          "bg",
        ]
      }
      fontFamily="heading"
      fontSize={
        Array [
          4,
          4,
          5,
          5,
        ]
      }
      width="fit-content"
    >
      Filters
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M11.46 8.29a1 1 0 0 0-1.42 0l-3 3a1 1 0 0 0 0 1.42l3 3a1 1 0 0 0 1.42 0a1 1 0 0 0 0-1.42L9.16 12l2.3-2.29a1 1 0 0 0 0-1.42zm3.2 3.71L17 9.71a1 1 0 0 0-1.42-1.42l-3 3a1 1 0 0 0 0 1.42l3 3a1 1 0 0 0 1.42 0a1 1 0 0 0 0-1.42z\\" fill=\\"#F3F0F0\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
    </h2>
  </span>
  <form
    className="sc-fzqNJr kTEIeG"
  >
    <label
      className="sc-fzokOt eSGGfa"
      color={
        Array [
          "bg",
        ]
      }
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
        ]
      }
      fontWeight="regular"
      for="address-filter"
      htmlFor="address-filter"
    >
      Alamat
      <input
        className="sc-fzoLsD eKoCBG"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        id="address-filter"
        onChange={[Function]}
        placeholder="Alamat rumah"
      />
    </label>
    <label
      className="sc-fzokOt eSGGfa"
      color={
        Array [
          "bg",
        ]
      }
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
        ]
      }
      fontWeight="regular"
      for="location-filter"
      htmlFor="location-filter"
    >
      Daerah
    </label>
    <div
      className="sc-AxheI kudZHY"
      width={1}
    >
      <label
        className="sc-AxmLO gpLzHo"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        for="location-filter"
        htmlFor="location-filter"
      >
        
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
            }
          }
          focusable={false}
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "transform": "rotate(360deg)",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        />
        <input
          id="location-filter"
          onChange={[Function]}
          type="checkbox"
        />
      </label>
    </div>
    <label
      className="sc-fzokOt eSGGfa"
      color={
        Array [
          "bg",
        ]
      }
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
        ]
      }
      fontWeight="regular"
      for="sale-type-filter"
      htmlFor="sale-type-filter"
    >
      Tipe penjualan property
    </label>
    <div
      className="sc-AxheI kudZHY"
      width={1}
    >
      <label
        className="sc-AxmLO gpLzHo"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        for="sale-type-filter"
        htmlFor="sale-type-filter"
      >
        
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
            }
          }
          focusable={false}
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "transform": "rotate(360deg)",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        />
        <input
          id="sale-type-filter"
          onChange={[Function]}
          type="checkbox"
        />
      </label>
    </div>
    <label
      className="sc-fzokOt eSGGfa"
      color={
        Array [
          "bg",
        ]
      }
      fontFamily="body"
      fontSize={
        Array [
          1,
          1,
          2,
        ]
      }
      fontWeight="regular"
      for="property-type-filter"
      htmlFor="property-type-filter"
    >
      Tipe property
    </label>
    <div
      className="sc-AxheI kudZHY"
      width={1}
    >
      <label
        className="sc-AxmLO gpLzHo"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        for="property-type-filter"
        htmlFor="property-type-filter"
      >
        
        <svg
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
            }
          }
          focusable={false}
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "transform": "rotate(360deg)",
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        />
        <input
          id="property-type-filter"
          onChange={[Function]}
          type="checkbox"
        />
      </label>
    </div>
    <div
      className="sc-fzoyAV lgNqAT"
    >
      <h3
        className="sc-AxiKw fkFCOK"
        color={
          Array [
            "bg",
          ]
        }
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
          ]
        }
        fontWeight="regular"
      >
        Luas rumah
      </h3>
      <div
        className="sc-fzoXzr bhBBAA"
        color={
          Array [
            "bg",
          ]
        }
        display="flex"
        fontFamily="body"
        fontSize={
          Array [
            2,
            2,
            3,
          ]
        }
        fontWeight="regular"
        width={1}
      >
        <label
          className="sc-fzokOt hAKGIV"
          for="area-min"
          htmlFor="area-min"
        >
          <input
            className="sc-fzoLsD KnxUl"
            color={
              Array [
                "bg",
              ]
            }
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            id="area-min"
            onChange={[Function]}
            placeholder="Min"
            type="number"
            value=""
            width={
              Array [
                72,
                72,
                96,
              ]
            }
          />
        </label>
        <span>
          -
        </span>
        <label
          className="sc-fzokOt hAKGIV"
          for="area-max"
          htmlFor="area-max"
        >
          <input
            className="sc-fzoLsD KnxUl"
            color={
              Array [
                "bg",
              ]
            }
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            fontWeight="bold"
            id="area-max"
            onChange={[Function]}
            placeholder="Max"
            type="number"
            value={0}
            width={
              Array [
                72,
                72,
                96,
              ]
            }
          />
        </label>
        <span>
          MÂ²
        </span>
      </div>
    </div>
    <div
      className="sc-fzoyAV sc-fzoLag fjoIQy"
    >
      <h3
        className="sc-AxiKw fkFCOK"
        color={
          Array [
            "bg",
          ]
        }
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
          ]
        }
        fontWeight="regular"
      >
        Harga rumah
      </h3>
      <div
        className="sc-fzoXzr cazspI"
        width={1}
      >
        <div
          className="sc-fzoXzr hgZeOg"
          color={
            Array [
              "bg",
            ]
          }
          display="flex"
          fontFamily="body"
          fontSize={
            Array [
              2,
              2,
              3,
            ]
          }
          fontWeight="regular"
          width={1}
        >
          <span>
            Dari Rp
          </span>
          <label
            className="sc-fzokOt hAKGIV"
            for="price-min"
            htmlFor="price-min"
          >
            <input
              className="sc-fzoLsD edxtsp"
              color={
                Array [
                  "bg",
                ]
              }
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              id="price-min"
              onChange={[Function]}
              placeholder="dari harga"
              type="number"
              value=""
              width={
                Array [
                  100,
                  100,
                  128,
                ]
              }
            />
          </label>
          <div
            className="sc-AxheI loDikT"
            width="fit-content"
          >
            <label
              className="sc-AxmLO gpLzHo"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              for="price-min-accumulator"
              htmlFor="price-min-accumulator"
            >
              
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
              <input
                id="price-min-accumulator"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </div>
        </div>
        <div
          className="sc-fzoXzr hgZeOg"
          color={
            Array [
              "bg",
            ]
          }
          display="flex"
          fontFamily="body"
          fontSize={
            Array [
              2,
              2,
              3,
            ]
          }
          fontWeight="regular"
          width={1}
        >
          <span>
            Sampai Rp
          </span>
          <label
            className="sc-fzokOt hAKGIV"
            for="price-max"
            htmlFor="price-max"
          >
            <input
              className="sc-fzoLsD edxtsp"
              color={
                Array [
                  "bg",
                ]
              }
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              id="price-max"
              onChange={[Function]}
              placeholder="Sampai harga"
              type="number"
              value={0}
              width={
                Array [
                  100,
                  100,
                  128,
                ]
              }
            />
          </label>
          <div
            className="sc-AxheI loDikT"
            width="fit-content"
          >
            <label
              className="sc-AxmLO gpLzHo"
              fontFamily="body"
              fontSize={
                Array [
                  1,
                  1,
                  2,
                  2,
                ]
              }
              fontWeight="bold"
              for="price-max-accumulator"
              htmlFor="price-max-accumulator"
            >
              
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<path d=\\"M17 9.17a1 1 0 0 0-1.41 0L12 12.71L8.46 9.17a1 1 0 0 0-1.41 0a1 1 0 0 0 0 1.42l4.24 4.24a1 1 0 0 0 1.42 0L17 10.59a1 1 0 0 0 0-1.42z\\" fill=\\"#2E282A\\"/>",
                  }
                }
                focusable={false}
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style={
                  Object {
                    "transform": "rotate(360deg)",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              />
              <input
                id="price-max-accumulator"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    className="sc-fznyAO fUmhzH"
    fontFamily="body"
    fontSize={
      Array [
        1,
        1,
        2,
        2,
      ]
    }
    fontWeight="bold"
    width={1}
  >
    Cari Sekarang
  </button>
</div>
`;
